packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -130,14 +192,317 @@ function isEmptyConfig({
   );
 }
 
-export class IrisGrid extends Component {
+type Settings = {
+  defaultDateTimeFormat?: string;
+  defaultDecimalFormatOptions?: {
+    defaultFormatString?: string;
+  };
+  defaultIntegerFormatOptions?: {
+    defaultFormatString?: string;
+  };
+  formatter?: FormattingRule[];
+  timeZone?: string;
+  showTimeZone?: boolean;
+  showTSeparator?: boolean;
+  truncateNumbersWithPound?: boolean;
+  disableMoveConfirmation?: boolean;
+  showSystemBadge?: boolean;
+};
+
+export type QuickFilter = {
+  text: string;
+  filter: FilterCondition | null;
+};
+
+export type AdvancedFilter = {
+  filter: FilterCondition | null;
+  options: Options;
+};
+
+export type Action = {
+  action: () => void;
+  shortcut: Shortcut;
+};
+
+export type OptionItem = {
+  type: OptionType;
+  title: string;
+  subtitle?: string;
+  icon?: IconDefinition;
+  isOn?: boolean;
+  onChange?: () => void;
+};
+
+export interface UITotalsTableConfig extends TotalsTableConfig {
+  operationOrder: AggregationOperation[];
+  showOnTop: boolean;
+}
+
+export type InputFilter = { name: string; type: string; value: string };
+export interface IrisGridProps {
+  children: React.ReactNode;
+  advancedFilters: Map<number, AdvancedFilter>;
+  advancedSettings: Map<AdvancedSettingsType, boolean>;
+  alwaysFetchColumns: string[];
+  isFilterBarShown: boolean;
+  applyInputFiltersOnInit: boolean;
+  conditionalFormats: SidebarFormattingRule[];
+  customColumnFormatMap: Map<string, FormattingRule>;
+  movedColumns: { from: number; to: number }[];
+  movedRows: { from: number; to: number }[];
+  inputFilters: InputFilter[];
+  customFilters: unknown[];
+  model: IrisGridProxyModel;
+  onCreateChart: (
+    settings: ChartBuilderSettings,
+    model: IrisGridProxyModel
+  ) => void;
+  onColumnSelected: (column: Column) => void;
+  onError: (error: unknown) => void;
+  onDataSelected: (index: number, map: unknown) => void;
```

Comment:
We should introduce a `type ColumnName = string` so that we can be more descriptive in places where we expect a `ColumnName`.
```suggestion
  onDataSelected: (index: ModelIndex, map: Record<ColumnName, unknown>) => void;
```